src/main/scala/org/constellation/primitives/Schema.scala

```diff
@@ -229,7 +237,24 @@ object Schema {
     lastTxRef: LastTransactionRef,
     fee: Option[Long] = None,
     salt: Long = Random.nextLong()
-  ) extends Signable
+  ) extends Signable {
+
+    override def getEncoding = {
+      val amountLengthString = amount.toString.length.toString
+      val feeLengthString = fee.getOrElse(0L).toString.length
+      val saltLengthString = amount.toString.length.toString
+      val args = Seq(
+        amountLengthString,
+        amount.toString,
+        lastTxRef.getEncoding,
+        feeLengthString.toString,
+        fee.getOrElse(0L).toString,
+        saltLengthString.toString,
+        salt.toString
+      )
+      runLengthEncoding(args: _*)
```

Comment:
all good but you have length already inside the `runLengthEncoding`:
```
s"$acc${hash.length}$hash")
```
so you should pass only `Seq(amount.toString, lastTxRef.getEncoding, fee.getOrElse(0L).toString, salt.toString)`

---

I can actually see you're using `lastTxRef.getEncoding` so it will be encoded twice. If you still want to use that, you should use `s"$acc$hash")` instead of `s"$acc${hash.length}$hash")` and then pass all lengths manually by yourself as you did

Reply:
Ooooooooh, nice catch!